numalogic/transforms/_movavg.py

```diff
@@ -114,32 +111,6 @@ def transform(self, input_: npt.NDArray[float], **__):
         ------
             InvalidDataShapeError: if input array is not single featured
         """
-        _allow_only_single_feature(input_)
-
-        # alpha is the weight given to the latest element
-        alpha = 1.0 - self.beta
-        n = len(input_)
-
-        theta = input_.reshape(-1, 1)
-        theta_tril = np.multiply(theta.T, np.tril(np.ones((n, n))))
-        powers = np.arange(1, n + 1).reshape(-1, 1)
-
-        # Calculate increasing powers of beta of the form,
-        # [beta, beta**2, .., beta**n]
-        beta_powers = np.power(self.beta, powers)
-
-        # Calculate the array of reciprocals of beta powers of form,
-        # [beta**(-1), beta**(-2), .., beta**(-n)]
-        beta_arr_inv = np.reciprocal(beta_powers)
-
-        # Calculate the summation of the ratio between (theta(i) / beta**i),
-        # [ theta(1)/beta, sum(theta(1)/beta, theta(2)/beta**2), .., ]
-        theta_beta_ratio = theta_tril @ beta_arr_inv
-
-        # Elemental multiply with beta powers
-        exp_avg = alpha * np.multiply(beta_powers, theta_beta_ratio)
-        if not self.bias_correction:
-            return exp_avg
-
-        # Calculate array of 1 / (1 - beta**i) values
-        return np.divide(exp_avg, 1.0 - beta_powers)
+        x_df = pd.DataFrame(input_, dtype=np.float32, copy=True)
+        x_smoothed = x_df.ewm(alpha=self.alpha).mean().to_numpy(dtype=np.float32)
```

Comment:
I think we should return without a mean.

Reply:
df.ewm does not return a real valued number. It just returns an object. We need to aggregate in the end.

Reply:
gotcha!